kvm/src/lib.rs

```diff
@@ -1195,6 +1349,21 @@ mod tests {
                     assert_eq!(data[0], 0);
                 }
                 VcpuExit::Hlt => {
+                    let dirty_pages = mem.map_and_fold(
+                        0,
+                        |(slot, memory_region)| {
+                            let bitmap = vm_fd
+                                .get_and_reset_dirty_page_bitmap(slot as u32, memory_region.size());
+                            match bitmap {
+                                Ok(v) => v
+                                    .iter()
+                                    .fold(0, |init, page| init + page.count_ones() as usize),
+                                Err(_) => 0,
+                            }
+                        },
+                        |dirty_pages, region_dirty_pages| dirty_pages + region_dirty_pages,
+                    );
+                    assert_ne!(dirty_pages, 0);
```

Comment:
Can this assert be more specific and assert that there are two dirty pages (one from loading the code, and one from the 'mov' in the source)?

Reply:
Done